components/Navbar/Navbar.tsx

```diff
@@ -0,0 +1,234 @@
+import * as React from 'react';
+import AppBar from '@mui/material/AppBar';
+import Box from '@mui/material/Box';
+import Toolbar from '@mui/material/Toolbar';
+import IconButton from '@mui/material/IconButton';
+import Typography from '@mui/material/Typography';
+import Menu from '@mui/material/Menu';
+import MenuIcon from '@mui/icons-material/Menu';
+import Container from '@mui/material/Container';
+import Avatar from '@mui/material/Avatar';
+import Tooltip from '@mui/material/Tooltip';
+import MenuItem from '@mui/material/MenuItem';
+import {
+  signOut, useSession,
+} from 'next-auth/react';
+import { useRouter } from 'next/router';
+
+const pages = ['Home'];
+
+const settings = [
+  'Logout',
+  'Profile',
+];
+
+type NavbarProps = {
+  children: React.ReactNode;
+}
+
+function ResponsiveAppBar({ children }: NavbarProps) {
+  const [
+    anchorElNav,
+    setAnchorElNav,
+  ] = React.useState<null | HTMLElement>(null);
+  const [
+    anchorElUser,
+    setAnchorElUser,
+  ] = React.useState<null | HTMLElement>(null);
+
+  const handleOpenNavMenu = (event: React.MouseEvent<HTMLElement>) => {
+    setAnchorElNav(event.currentTarget);
+  };
+  const handleOpenUserMenu = (event: React.MouseEvent<HTMLElement>) => {
+    setAnchorElUser(event.currentTarget);
+  };
+
+  const handleCloseNavMenu = () => {
+    setAnchorElNav(null);
+  };
+
+  const handleCloseUserMenu = () => {
+    setAnchorElUser(null);
+  };
+
+  const { data: session } = useSession();
+  const router = useRouter();
+
+  const handleSettingClick = (setting: string) => {
+    if (setting === 'Logout') {
+      signOut();
+      router.push('/');
+    } else if (setting === 'Profile') {
+      handleCloseUserMenu();
+      router.push('/profile');
+    }
+  };
+
+  const handlePageClick = (page: string) => {
+    if (page === 'Home') {
+      handleCloseNavMenu();

```

Comment:
That's not needed if you add onClick prop to Menu

Reply:
Also you could do it other way.
const pages could be array of objects e.g. 
[{path: '/', name: 'Home'}, ...]
then you wouldn't need to check what's page name, you would just pass the path string as argument to this function and use router push with it.
Or evene get rid of this function altogether and just use router.push(path) where you map the pages array in JSX